src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -619,4 +1111,88 @@ private function transformJsonPathRegex(string $pattern): string
 
         return $result;
     }
+
+    private function isArrayOrObject(mixed $value): bool
+    {
+        return \is_array($value) || $value instanceof \stdClass;
+    }
+
+    private function hasKey(mixed $value, string $key): bool
+    {
+        if (\is_array($value)) {
+            return \array_key_exists($key, $value);
+        }
+
+        return $value instanceof \stdClass && property_exists($value, $key);
+    }
+
+    private function getValue(mixed $value, string $key): mixed
+    {
+        if (\is_array($value) && \array_key_exists($key, $value)) {
+            return $value[$key];
+        }
+
+        if ($value instanceof \stdClass && property_exists($value, $key)) {
+            return $value->$key;
+        }
+
+        return null;
+    }
+
+    private function getValues(mixed $value): array
+    {
+        if (\is_array($value)) {
+            return array_values($value);
+        }
+
+        if ($value instanceof \stdClass) {
+            return array_values(get_object_vars($value));
+        }
+
+        return [];
+    }
+
+    private function getCount(mixed $value): int
+    {
+        if (\is_array($value)) {
+            return \count($value);
+        }
+
+        if ($value instanceof \stdClass) {
+            return \count(get_object_vars($value));
+        }
+
+        return 0;
```

Comment:
I'm not even sure this deserves a helper method:
```suggestion
        return \count((array) $value);
```